cips/cip-30.md

```diff
@@ -0,0 +1,108 @@
+| cip           | 30                                                                                         |
+|---------------|--------------------------------------------------------------------------------------------|
+| title         | Prevent Auto-Claiming of Staking Rewards                                                   |
+| description   | This CIP proposes removing the automatic claiming of staking rewards when modifying delegations. Instead, rewards are to be stored until manually claimed. |
+| author        | Dean Eigenmann ([@decanus](https://github.com/decanus)), Marko Baricevic ([@tac0turtle](https://github.com/tac0turtle)) |
+| discussions-to |                                 |
+| status        | Draft                                                                                      |
+| type          | Standards Track                                                                            |
+| category      | Core                                                                                       |
+| created       | 2025-01-27                                                                                 |
+
+## Abstract
+
+Currently, when delegators modify their staking positions (e.g., redelegate or undelegate), the distribution module automatically claims all accrued staking rewards. This CIP proposes removing that automatic functionality, thereby allowing users to intentionally choose *when* to claim their rewards. Specifically, the distribution module will no longer auto-claim rewards on delegation-share changes. Instead, it will store the accrued rewards until a user explicitly calls `MsgWithdrawDelegatorReward`. This avoids unintended reward-claiming events that can trigger immediate tax or regulatory obligations.
```

Comment:
seems like a simple enough change, do we have data on the rate of redelegations or undelegations or how much demand there is for this change? I couldn't think of any firm blockers to this, but its been a while since I dug through the maze that is the distribution and staking module logic.

separate to this limitted change,

>This is a simple change, if we wanted something more complex we could propose an LST like design

could we do, or has any sdk based chain done, claimless auto compounding, redelegation, undelegation without creating an LST? presumably this requires a significant change

Reply:
yea i can get the data for the last few months. 

we have explored this, its a deeper change in the distribution module. I can look into this as an alternative.

Reply:
so changing to be native compounding can be done, its not that hard. The main change is the UX around, users will need to undelegate to claim rewards. This requires a conversation on UX before moving forward. The nice part is that we would reduce state machine complexity per block significantly.